nomad/util.go

```diff
@@ -196,6 +196,34 @@ func shuffleStrings(list []string) {
 	}
 }
 
+// partitionAll splits a slice of strings into a slice of slices of strings, each with a max
+// size of `size`. All entries from the original slice are preserved. The last slice may be
+// smaller than `size`. The input slice is unmodified
+func partitionAll(size int, xs []string) [][]string {
+	out := make([][]string, 0)
+	if size < 1 {
+		return append(out, xs)
+	}
+
+	got, part, i, j := 0, 0, 0, 0
```

Comment:
Might be worth describing the algorithm in comments.   some of these variables are only used in the loop, so you probably can simplify it a bit with something like:

```golang
 	for i := 0; i < len(xs); i += size {
	        j := minInt(len(xs), i+size)
		out = append(out, xs[i:j])
	}
```

I might inline minInt too but that's nitpicking.